### ISSUE TITLE
[FR] Allow for pipeline to be defined in a js file

### ISSUE THREAD

**Issue by rafamel** [pikapkg/cli#2]
Hi, congrats on the project!

It would be great to be able to define the pack pipeline in a different js file rather than in `package.json`. I'm suggesting something like `pack.config.js` - [liftoff](https://github.com/js-cli/js-liftoff) could simplify implementation. Is that something we might get in the future?

Rafa.

**Comment by FredKSchott** [comment]
Hey all, moving this here as a general CLI discussion.

I think the direction that the project has been moving has led us to the need for a single standard `pika.json` config file format. This would be something where you could define configuration for @pika/web, @pika/pack, and any future pika commands. Here's a rough draft of what this would look like:

```js
// pika.json
{
  "package": {
    "out": "dist/", /* All CLI arguments supported as config */
    "pipeline": [ /* currently: package.json["@pika/pack"]["pipeline'] */ ],
  },
  "web": {
    "clean": true, /* All CLI arguments supported as config */
    "dependencies": [ /* currently: package.json["@pika/web"]["webDependencies'] */ ],
  }
}
```

But I'm actually kind of sad to lose "webDependencies" in the "package.json", since that goes so well with "dependencies" & "devDependencies". Maybe we leave that one value in the package.json at the top-level, or support it in both places?

I also see a lot of projects moving to TOML, which I like.  +1 for adding support for both JSON & TOML config file formats.

**Comment by NicoCevallos** [comment]
Hi! I just back to use pika again and feel sad to see this is pending :(

@FredKSchott do you have any plan to do that?

**Comment by FredKSchott** [comment]
Yes, although renaming @pika/web to Snowpack complicates that original plan a bit. We'll probably just give them each their own config (`snowpack.config.json` and `pika-pack.config.json` or something similar, since you'd rarely use the two together.)
